### PR TITLE
Update utils.py

### PR DESCRIPTION
**Pull request by shawnluo-528**
To fix the MemoryError issue when pack a big file.
  File "c:\python27\lib\site-packages\PyInstaller\building\utils.py", line 333, in cacheDigest
    data = open(fnm, "rb").read()
MemoryError

### CODE REVIEW

File: PyInstaller/building/utils.py [hunk 1]
```diff
@@ -337,9 +337,19 @@ def checkCache(fnm, strip=False, upx=False, dist_nm=None):
...
 
+def get_md5(fnm):
+    hasher = hashlib.md5()
+    with open(fnm, 'rb') as fh:
+        while True:
+            data = fh.read(2048)    # 2048 each iteration
+            if not data:
+                break
+            hasher.update(data)
+    return hasher
```
Comment (xoviat) [inline PyInstaller/building/utils.py]:
```python
def md5(fname):
    hash_md5 = hashlib.md5()
    with open(fname, "rb") as f:
        for chunk in iter(lambda: f.read(4096), b""):
            hash_md5.update(chunk)
    return hash_md5.hexdigest()
```

Comment (shawnluo-528) [reply PyInstaller/building/utils.py]: that's good

**Comment by htgoebel** [comment]
Thank you for this fix, which is a great small improvement :-) Just two minot changes:
- Using only 4096 bytes per read seems to be a bit greedy. In PyInstaller/archive/writers.py we use chunks of 16*1024 bytes. I suggest doing the same here.
- Please put the code directly into `cacheDigest()`, no into a (sub-) function. md5 is not used elsewhere  in PyInstaller.
Thanks.

@xoviat This use of iter is quite cool. I did not know it yet :-)

**Comment by htgoebel** [comment]
@shawnluo-528 Please update the pull-request as requested. Thanks.

**Pull request merged**